File: backend/member-service/src/test/java/com/example/memberservice/api/internal/InternalApiFeignConfigTest.java
package com.example.memberservice.api.internal;

import static org.junit.jupiter.api.Assertions.assertEquals;

import feign.RequestTemplate;
import org.junit.jupiter.api.Test;

class InternalApiFeignConfigTest {

    @Test
    void interceptor_attachesInternalToken() {
        RequestTemplate template = new RequestTemplate();
        new InternalApiFeignConfig().internalApiTokenInterceptor("secret-token").apply(template);
        assertEquals(java.util.List.of("secret-token"), java.util.List.copyOf(template.headers().get("X-Internal-Token")));
    }
}

File: backend/auth-service/src/test/kotlin/com/example/authservice/api/internal/InternalApiFeignConfigTest.kt
package com.example.authservice.api.internal

import feign.RequestTemplate
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

class InternalApiFeignConfigTest {

    @Test
    fun interceptor_attachesInternalToken() {
        val template = RequestTemplate()
        InternalApiFeignConfig().internalApiTokenInterceptor("secret-token").apply(template)
        assertEquals(listOf("secret-token"), template.headers()["X-Internal-Token"]!!.toList())
    }
}
